google/cloud/compute/CMakeLists.txt

```diff
@@ -292,11 +374,59 @@ install(
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/google_cloud_cpp_compute"
     COMPONENT google_cloud_cpp_development)
 
-external_googleapis_install_pc("google_cloud_cpp_compute_protos")
+unset(pkgconfig_deps)
+list(APPEND pkgconfig_deps "google_cloud_cpp_cloud_extended_operations_protos")
+compute_pkgconfig(google_cloud_cpp_compute_internal_protos DEPS
+                  ${pkgconfig_deps})
 
-# google-cloud-cpp::sql must be defined before we can add the samples.
+unset(pkgconfig_deps)
+list(
+    APPEND
+    pkgconfig_deps
+    "google_cloud_cpp_compute_internal_protos"
+    "google_cloud_cpp_api_annotations_protos"
+    "google_cloud_cpp_api_client_protos"
+    "google_cloud_cpp_api_field_behavior_protos")
+foreach (dir IN LISTS service_dirs operation_service_dirs)
+    unset(GOOGLE_CLOUD_CPP_PC_REQUIRES)
+    unset(DEPS)
+    if (NOT "${dir}" STREQUAL "v1/")
+        string(REPLACE "/v1/" "" short_dir "${dir}")
+        compute_pkgconfig("google_cloud_cpp_compute_${short_dir}_protos" DEPS
+                          ${pkgconfig_deps})
+    endif ()
+endforeach ()
+
+unset(pkgconfig_deps)
+list(APPEND pkgconfig_deps "google_cloud_cpp_rest_internal"
+     "google_cloud_cpp_rest_protobuf_internal" "google_cloud_cpp_grpc_utils"
```

Comment:
comment: There is a bit of redundancy in the pkg config modules + library deps above... but I am not sure it is worth addressing.

It might be possible to do the pkg config stuff in the same loop that defines the libraries above.... but I haven't thought it through. So feel free to ignore this comment.